frontend/src/layout.ts

```diff
@@ -55,19 +73,44 @@ export class StatefulLayout {
     }
   }
 
-  nodeColor(eventType: string, hue?: number) {
-    if (!this.colors.has(eventType)) {
-      const hash = [...eventType].reduce(
-        (acc, char) => char.charCodeAt(0) + ((acc << 1) - acc),
-        2
-      )
-      const color = `hsl(${hash % (hue ? hue : 360)},50%,50%)`
-      this.colors.set(eventType, color)
-    }
-    return this.colors.get(eventType)
+  nodeColor(eventType: string) {
+    // if (!this.colors.has(eventType)) {
+    //   const hash = [...eventType].reduce(
+    //     (acc, char) => char.charCodeAt(0) + ((acc << 1) - acc),
+    //     2
+    //   )
+    //   const color = `hsl(${hash % (hue ? hue : 360)},50%,50%)`
+    //   this.colors.set(eventType, color)
+    return this.themeMap.get(eventType).Color
```

Comment:
There is an error regarding this line: layout.ts:84 `Uncaught(in promise) TypeError: Cannot read properties of undefined (reading 'Color')`

Reply:
> There is an error regarding this line: layout.ts:84 `Uncaught(in promise) TypeError: Cannot read properties of undefined (reading 'Color')`

There was an issue with the event generator. So it works with database dump and looks very nice. The labels are also displayed atm. Since we have the legend with different shapes and colors, there is no need for labels for the nodes.